### PR TITLE
virsh_save: add save with --xml option test

### PR DESCRIPTION
**Pull request by yanqzhan**
Add save with --xml option test.

Signed-off-by: Yanqiu Zhang <yanqzhan@redhat.com>

### CODE REVIEW

**Comment by yanqzhan** [comment]
To auto case: VIRT-16905 - [save][--xml] Supply an alternative XML file for use on the restored guest

Local test pass:
#avocado run --vt-type libvirt --vt-machine-type q35 virsh.save..xml_option
JOB ID     : 53619a91a8e4046d63dc7b52908afdc010149d03
JOB LOG    : /root/avocado/job-results/job-2022-05-12T08.56-53619a9/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.save.normal_test.normal_option.xml_option.no_progress: PASS (27.14 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 28.15 s

File: libvirt/tests/src/virsh_cmd/domain/virsh_save.py [hunk 1]
```diff
@@ -50,6 +52,26 @@ def run(test, params, env):
             test.cancel("API acl test not supported in current"
                         " libvirt version.")
 
+    def alter_xml(vm_name):
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_save.py:55]: method name change from alter_xml to create_alter_xml?

Comment (yanqzhan) [reply libvirt/tests/src/virsh_cmd/domain/virsh_save.py]: Done. Thanks!
 (1/1) type_specific.io-github-autotest-libvirt.virsh.save.normal_test.normal_option.xml_option.no_progress: -PASS (40.00 s)

**Review by chunfuwen — APPROVED** [review]
LGTM

**Review by cliping — APPROVED** [review]
LGTM

**Pull request merged**